commit 34e97345715fc072cd70087cbc854097c9126776
Author: bertin <macportsuser@macports.org>
Date:   Sun May 25 22:22:21 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-allow-archcheck-override.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 25eac883f..4fa0f944e 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -2969,6 +2969,17 @@ proc _mportcheck_known_fail {options portinfo} {
 
 ### _mportexec is private; may change without notice
 
+proc arch_supported {mport checked} {
+    set options [ditem_key $mport options]
+    if {[dict exists $options ports_force] && [dict get $options ports_force]} {
+        if {${checked} != 0} {
+            ui_debug "Ignoring check_supported_archs failure because of -f"
+        }
+        return 1
+    }
+    return [expr ${checked} == 0]
+}
+
 proc _mportexec {target mport} {
     set portname [_mportkey $mport subport]
     macports::push_log $mport
@@ -2981,7 +2992,7 @@ proc _mportexec {target mport} {
     # version (and presence for use_xcode yes ports).
     if {![catch {$workername eval [list check_variants $target]} result] && $result == 0 &&
         (![macports::_target_needs_toolchain $workername $target] || (![catch {$workername eval [list _check_xcode_version]} result] && $result == 0)) &&
-        ![catch {$workername eval [list check_supported_archs]} result] && $result == 0 &&
+        ![catch {$workername eval [list check_supported_archs]} result] && [arch_supported $mport $result] &&
         ![catch {$workername eval [list eval_targets $target]} result] && $result == 0} {
         # If auto-clean mode, clean-up after dependency install
         global macports::portautoclean
@@ -3037,7 +3048,7 @@ proc mportexec {mport target} {
             return 1
         }
         # error out if selected arch(s) not supported by this port
-        if {[$workername eval [list check_supported_archs]] != 0} {
+        if {![arch_supported $mport [$workername eval [list check_supported_archs]]]} {
             if {$log_needs_pop} {
                 macports::pop_log
             }
@@ -4474,6 +4485,10 @@ proc mportdepends {mport {target {}} {recurseDeps 1} {skipSatisfied 1} {accDeps
                 && [lsearch -exact -nocase $depends_skip_archcheck $dep_portname] == -1} {
                 set check_archs 1
             }
+            if {[dict exists $options ports_force] && [dict get $options ports_force]} {
+                ui_debug "Force-disabling architecture checks because of -f"
+                set check_archs 0
+            }
 
             # need to open the portfile even if the dep is installed if it doesn't have the right archs
             set parse 0
